pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go

```diff
@@ -186,51 +181,61 @@ func makeClusters(
 			continue
 		}
 
-		name := service
-
-		if len(ref.Tags) > 0 {
-			name = envoy_names.GetSplitClusterName(service, splitCounter.getAndIncrement())
+		switch plugins_xds.InferProtocol(proxy.Routing, service) {
+		case core_mesh.ProtocolHTTP, core_mesh.ProtocolHTTP2:
+		default:
+			// We don't support splitting if at least one of the backendRefs is not HTTP
+			return nil
 		}
 
-		// The mesh tag is present here if this destination is generated
-		// from a cross-mesh MeshGateway listener virtual outbound.
-		// It is not part of the service tags.
-		if mesh, ok := ref.Tags[mesh_proto.MeshTag]; ok {
-			// The name should be distinct to the service & mesh combination
-			name = fmt.Sprintf("%s_%s", name, mesh)
-		}
+		clusterName := getClusterName(ref, sc)
+		isExternalService := plugins_xds.HasExternalService(proxy.Routing, service)
+		allTags := envoy_tags.Tags(ref.Tags).WithTags(mesh_proto.ServiceTag, ref.Name)
 
-		// We assume that all the targets are either ExternalServices or not
-		// therefore we check only the first one
-		var isExternalService bool
-		if endpoints := proxy.Routing.OutboundTargets[service]; len(endpoints) > 0 {
-			isExternalService = endpoints[0].IsExternalService()
-		}
-		if endpoints := proxy.Routing.ExternalServiceOutboundTargets[service]; len(endpoints) > 0 {
-			isExternalService = true
+		if existingClusterName, ok := clusterCache[allTags.String()]; ok {
+			// cluster already exists, so adding only split
+			split = append(split, plugins_xds.NewSplitBuilder().
+				WithClusterName(existingClusterName).
+				WithWeight(uint32(ref.Weight)).
+				WithExternalService(isExternalService).
+				Build())
+			continue
 		}
 
-		allTags := envoy_tags.Tags(ref.Tags).WithTags(mesh_proto.ServiceTag, ref.Name)
-		cluster := envoy_common.NewCluster(
-			envoy_common.WithService(service),
-			envoy_common.WithName(name),
-			envoy_common.WithWeight(uint32(ref.Weight)),
-			envoy_common.WithTags(allTags.WithoutTags(mesh_proto.MeshTag)),
-			envoy_common.WithExternalService(isExternalService),
-		)
-
-		if mesh, ok := ref.Tags[mesh_proto.MeshTag]; ok {
-			cluster.SetMesh(mesh)
-		}
+		clusterCache[allTags.String()] = clusterName
 
-		if name, ok := clusterCache[allTags.String()]; ok {
-			cluster.SetName(name)
-		} else {
-			clusterCache[allTags.String()] = cluster.Name()
-		}
+		split = append(split, plugins_xds.NewSplitBuilder().
+			WithClusterName(clusterName).
+			WithWeight(uint32(ref.Weight)).
+			WithExternalService(isExternalService).
+			Build())
+
+		servicesAcc.Add(plugins_xds.NewClusterBuilder().
+			WithService(service).
+			WithName(clusterName).
+			WithMesh(ref.Tags[mesh_proto.MeshTag]).
```

Comment:
This depends on the empty value of the meshtag being ""... This feels too implicit, maybe we can keep it like before with checking `mesh, ok := ref.Tags[mesh_proto.MeshTag]; ok`?

Reply:
Yeah, it's probably to have it explicitly set. Fixed